cmd/tools_gotty.go

```diff
@@ -347,53 +321,25 @@ func gottyAttach(cobraCmd *cobra.Command, o *Options) error { //nolint: funlen
 		o.Global.TopologyFile,
 	)
 
-	// Get lab topology information
-	clabInstance, err := clabcore.NewClabFromTopologyFileOrLabName(
-		o.Global.TopologyFile,
-		o.Global.TopologyName,
-		o.Global.VarsFile,
-		o.Global.Runtime,
-		o.Global.DebugCount > 0,
-		o.Global.Timeout,
-		o.Global.GracefulShutdown,
-	)
+	c, err := clabcore.NewContainerLab(o.ToClabOptions()...)
 	if err != nil {
 		return err
 	}
 
-	labName := clabInstance.Config.Name
-	networkName := clabInstance.Config.Mgmt.Network
-
-	if networkName == "" {
-		networkName = "clab-" + labName
+	rt, ok := c.Runtimes[o.Global.Runtime]
```

Comment:
removed all the runtime initializer stuff in here because we already do that during NewContainerlab basically